### ISSUE TITLE
Implement the Save System and Main Menu.

### ISSUE THREAD

**Issue by alkalinethunder** [bitphoenixsoftware/the-peacenet#6]
In Unreal Engine as well as Old Peacenet, we had a main menu in the game.  This main menu essentially allowed the user to configure the game, and allowed them to manage their save files (or **Agent Identities.**)

Currently in the MonoGame port, this is not the case.  To help speed the port along, I think we should tackle this now that the world generator has been mostly ported over.

There are a few things we need to do to get this done, outlined below.

**Backend**

 - [x] Add a "profile", containing information about all the save files in the game.  This should be stored in local app data along with the already-existing `HypervisorConfig.json` file.  Perhaps we should use LiteDB for this profile to allow storing of other game data that isn't specific to a save file?  (Perhaps things lik API keys/OAuth2 tokens as well as a server list if we ever add multiplayer back?)
 - [x] Write a game loader/saver.  I think we should write our own format which works in tandem with the Greenlight Preloader, so that we can serialize `Asset` objects as asset IDs in the file, allowing asset references to be loaded back.  A JSON or Whoa serializer would simply serialize the object data which wouldn't be ideal in the case of an `Asset`;  nor do we want to manually deal with Asset IDs in-game.

**GUI**

 - [x] Add a main screen that holds all the main menu widgets, and show it after pre-loading is done instead of the desktop.
 - [x] Add the ability to create a new game - ideally the player should be able to name their Agent and thus affect world generation.
 - [x] Add the ability to manage and load existing Agents/save files.
 - [ ] Add a Settings GUI for setting screen resolution and other in-game settings.

**Comment by TheDarkBomber** [comment]
Hypervisor is the supervisor of a supervisor.

**Comment by alkalinethunder** [comment]
I'm aware of this.  VMs are run in Hypervisors... that was the point.  The player's in-game computer is just a VM allocated by the Agency within The Peacenet.

So effeectively the actual game executable which is running the world simulation would be the hypervisor running that VM, giving it access to the host system's hardware.

**Comment by TheDarkBomber** [comment]
oh so I guessed right

**Comment by alkalinethunder** [comment]
Profile/save manager is added to the game.  It supports multiple saves and stores information such as agent name, last played date and relative path to save file (relative to `%appdata%\roaming\Bit Phoenix Software\The Peacenet\saves` (and equivalent on Linux/macOS)).  Game will remove entries from the database whose files don't exist on disk, and soon will add new entries when save files are added to the saves folder that aren't in the database.

So basically the database just gives us an easy way to keep track of all the save files and store some basic info about them without having to fully read them.  We'd have to read into the file at some point though when "rebuilding" the database.

**Comment by alkalinethunder** [comment]
Ability to create a new game is in.  Decided to use `BinaryFormatter` for save serialization for now, might need to write some custom serializer for `Asset` objects.  Though, the game'll throw a fit if you try to serialize an object who's not `Serializable` so... since `Asset` isn't `Serializable` we should be able to enforce **not** storing asset objects in save objects and instead only storing IDs.

**Comment by alkalinethunder** [comment]
Game will now let you load the 5 most recent save files in the menu.  It will also now save the game if MonoGame is exiting while Peacegate is active.

LiteDB is proving extremely useful for all of this.

**Comment by alkalinethunder** [comment]
Closed - monogame development is abandoned.
